<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Estimation of π₀ · MultipleTesting.jl</title><link rel="canonical" href="https://juliangehring.github.io/MultipleTesting.jl/stable/pi0.html"/><link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css" rel="stylesheet" type="text/css"/><link href="https://fonts.googleapis.com/css?family=Lato|Roboto+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link href="assets/documenter.css" rel="stylesheet" type="text/css"/></head><body><nav class="toc"><h1>MultipleTesting.jl</h1><select id="version-selector" onChange="window.location.href=this.value" style="visibility: hidden"></select><form class="search" id="search-form" action="search.html"><input id="search-query" name="q" type="text" placeholder="Search docs"/></form><ul><li><a class="toctext" href="index.html">Home</a></li><li><span class="toctext">Library</span><ul><li><a class="toctext" href="adjustment.html">Adjustment of p-Values</a></li><li><a class="toctext" href="combination.html">Combination of p-Values</a></li><li class="current"><a class="toctext" href="pi0.html">Estimation of π₀</a><ul class="internal"><li><a class="toctext" href="#Methods-1">Methods</a></li><li><a class="toctext" href="#Types-1">Types</a></li></ul></li><li><a class="toctext" href="higher-criticism.html">Higher Criticism</a></li><li><a class="toctext" href="models.html">Modelling of p-Value Distributions</a></li></ul></li></ul></nav><article id="docs"><header><nav><ul><li>Library</li><li><a href="pi0.html">Estimation of π₀</a></li></ul></nav><hr/><div id="topbar"><span>Estimation of π₀</span><a class="fa fa-bars" href="#"></a></div></header><h1><a class="nav-anchor" id="Estimation-of-π-1" href="#Estimation-of-π-1">Estimation of π₀</a></h1><h2><a class="nav-anchor" id="Methods-1" href="#Methods-1">Methods</a></h2><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.estimate_pi0" href="#MultipleTesting.estimate_pi0"><code>MultipleTesting.estimate_pi0</code></a> — <span class="docstring-category">Function</span>.</div><div><pre><code class="language-none">estimate_pi0(PValues, Pi0Estimator)</code></pre><p>Estimate π₀, the fraction of tests under the null hypothesis</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, StoreyBootstrap())
0.0
julia&gt; estimate_pi0(pvals, FlatGrenander())
0.42553191489361697</code></pre><pre><code class="language-julia-repl">julia&gt; subtypes(Pi0Estimator)
10-element Array{Union{DataType, UnionAll},1}:
 MultipleTesting.BUM
 MultipleTesting.CensoredBUM
 MultipleTesting.ConvexDecreasing
 MultipleTesting.FlatGrenander
 MultipleTesting.LeastSlope
 MultipleTesting.Oracle
 MultipleTesting.RightBoundary
 MultipleTesting.Storey
 MultipleTesting.StoreyBootstrap
 MultipleTesting.TwoStep</code></pre><p><strong>See also</strong></p><p><code>Pi0Estimator</code>s:</p><p><a href="pi0.html#MultipleTesting.Storey"><code>Storey</code></a> <a href="pi0.html#MultipleTesting.StoreyBootstrap"><code>StoreyBootstrap</code></a> <a href="pi0.html#MultipleTesting.LeastSlope"><code>LeastSlope</code></a> <a href="pi0.html#MultipleTesting.Oracle"><code>Oracle</code></a> <a href="pi0.html#MultipleTesting.TwoStep"><code>TwoStep</code></a> <a href="pi0.html#MultipleTesting.RightBoundary"><code>RightBoundary</code></a> <a href="pi0.html#MultipleTesting.CensoredBUM"><code>CensoredBUM</code></a> <a href="pi0.html#MultipleTesting.BUM"><code>BUM</code></a> <a href="pi0.html#MultipleTesting.FlatGrenander"><code>FlatGrenander</code></a> <a href="pi0.html#MultipleTesting.ConvexDecreasing"><code>ConvexDecreasing</code></a></p></div></section><h2><a class="nav-anchor" id="Types-1" href="#Types-1">Types</a></h2><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.Storey" href="#MultipleTesting.Storey"><code>MultipleTesting.Storey</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Storey&#39;s π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, Storey())
0.22222222222222224

julia&gt; estimate_pi0(pvals, Storey(0.4))
0.33333333333333337</code></pre><p><strong>References</strong></p><p>Storey, J.D., Taylor, J.E., and Siegmund, D. (2004). Strong control, conservative point estimation and simultaneous conservative consistency of false discovery rates: a unified approach. Journal of the Royal Statistical Society: Series B (Statistical Methodology) 66, 187–205.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.StoreyBootstrap" href="#MultipleTesting.StoreyBootstrap"><code>MultipleTesting.StoreyBootstrap</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Storey&#39;s closed-form bootstrap π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, StoreyBootstrap())
0.0

julia&gt; estimate_pi0(pvals, StoreyBootstrap(0.1:0.1:0.9, 0.2))
0.0</code></pre><p><strong>References</strong></p><p>Robinson, D. (2016). Original Procedure for Choosing λ. http://varianceexplained.org/files/pi0boot.pdf</p><p>Storey, J.D., Taylor, J.E., and Siegmund, D. (2004). Strong control, conservative point estimation and simultaneous conservative consistency of false discovery rates: a unified approach. Journal of the Royal Statistical Society: Series B (Statistical Methodology) 66, 187–205.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.LeastSlope" href="#MultipleTesting.LeastSlope"><code>MultipleTesting.LeastSlope</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Least Slope (LSL) π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, LeastSlope())
1.0</code></pre><p><strong>References</strong></p><p>Benjamini, Y., and Hochberg, Y. (2000). On the Adaptive Control of the False Discovery Rate in Multiple Testing With Independent Statistics. Journal of Educational and Behavioral Statistics 25, 60–83.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.Oracle" href="#MultipleTesting.Oracle"><code>MultipleTesting.Oracle</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Oracle π₀</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, Oracle(0.5)) # a bit boring...
0.5</code></pre></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.TwoStep" href="#MultipleTesting.TwoStep"><code>MultipleTesting.TwoStep</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Two-step π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, TwoStep())
0.2

julia&gt; estimate_pi0(pvals, TwoStep(0.05, BenjaminiLiu()))
0.2
</code></pre><p><strong>References</strong></p><p>Benjamini, Y., Krieger, A.M., and Yekutieli, D. (2006). Adaptive linear step-up procedures that control the false discovery rate. Biometrika 93, 491–507.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.RightBoundary" href="#MultipleTesting.RightBoundary"><code>MultipleTesting.RightBoundary</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Right boundary π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, RightBoundary())
0.2127659574468085

julia&gt; estimate_pi0(pvals, RightBoundary(0.1:0.1:0.9))
0.25</code></pre><p><strong>References</strong></p><p>Liang, K., and Nettleton, D. (2012). Adaptive and dynamic adaptive procedures for false discovery rate control and estimation. Journal of the Royal Statistical Society: Series B (Statistical Methodology) 74, 163–182.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.CensoredBUM" href="#MultipleTesting.CensoredBUM"><code>MultipleTesting.CensoredBUM</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Censored Beta-Uniform Mixture (censored BUM) π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, CensoredBUM())
0.21052495526400936
</code></pre><p><strong>References</strong></p><p>Markitsis, A., and Lai, Y. (2010). A censored beta mixture model for the estimation of the proportion of non-differentially expressed genes. Bioinformatics 26, 640–646.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.BUM" href="#MultipleTesting.BUM"><code>MultipleTesting.BUM</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Beta-Uniform Mixture (BUM) π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, BUM())
0.22802795505154264</code></pre><p><strong>References</strong></p><p>Pounds, S., and Morris, S.W. (2003). Estimating the occurrence of false positives and false negatives in microarray studies by approximating and partitioning the empirical distribution of p-values. Bioinformatics 19, 1236–1242.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.FlatGrenander" href="#MultipleTesting.FlatGrenander"><code>MultipleTesting.FlatGrenander</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Flat Grenander π₀ estimator</p><p>Estimates π₀ by finding the longest constant interval in the Grenander estimator.</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, FlatGrenander())
0.42553191489361697</code></pre><p><strong>References</strong></p><p>Langaas, M., Lindqvist, B.H., and Ferkingstad, E. (2005). Estimating the proportion of true null hypotheses, with application to DNA microarray data. Journal of the Royal Statistical Society: Series B (Statistical Methodology) 67, 555–572.</p></div></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="MultipleTesting.ConvexDecreasing" href="#MultipleTesting.ConvexDecreasing"><code>MultipleTesting.ConvexDecreasing</code></a> — <span class="docstring-category">Type</span>.</div><div><p>Convex Decreasing π₀ estimator</p><p><strong>Examples</strong></p><pre><code class="language-julia-repl">julia&gt; pvals = PValues([0.001, 0.002, 0.01, 0.03, 0.5]);

julia&gt; estimate_pi0(pvals, ConvexDecreasing())
0.013007051336745304</code></pre><p><strong>References</strong></p><p>Langaas, M., Lindqvist, B.H., and Ferkingstad, E. (2005). Estimating the proportion of true null hypotheses, with application to DNA microarray data. Journal of the Royal Statistical Society: Series B (Statistical Methodology) 67, 555–572.</p></div></section><footer><hr/><a class="previous" href="combination.html"><span class="direction">Previous</span><span class="title">Combination of p-Values</span></a><a class="next" href="higher-criticism.html"><span class="direction">Next</span><span class="title">Higher Criticism</span></a></footer></article></body></html>
